### DISCUSSION TITLE
a lot of warnings of OneshotWorld

### DISCUSSION THREAD

**Discussion by snippet2021** [yasserfarouk/scml#91] [one-shot-open-discussion]
・environment
venv(python3.8) on wsl(ubuntu) on Windows10

I installed OneshotTemplate from website and installed scml environment. Following this video.
 https://youtu.be/HBQs89aTeLE

・phenomenon
When I run myagent.py, I get a lot of warnings like this.(more than 250)

>  warnings.warn(
(my directory)/.venv/lib/python3.8/site-packages/negmas/types/rational.py:53: NegmasDoubleAssignmentWarning: Entity lNEHT5lgrQs2Bzc1 (SingleAgreementAspirationAgent) is assigned preferences belonging to another entity (08SAA@1 of type SingleAgreementAspirationAgent)!!
 

・question
Is it safe to ignore this message? Or do I need to do something about it?

**Comment by yasserfarouk** [comment]
It is completely safe to ignore them. We are using a nonstandard way to assign ufuns that NegMAS warns about but it is tested and we know it does not cause any issues. We will remove this warning in the next patch.

for now you can avoid the warnings if you run python with the -W ignore command line argument

**Reply by snippet2021** [reply]
I see. Thank you very much for the swift reply.
